test.mjs

```diff
@@ -356,6 +356,23 @@ test('timeout', async (t) => {
   )
 })
 
+test('pending timeouts get cleaned up on close', async (t) => {
+  t.plan(2)
+  const rpc = new RPC(new PassThrough())
+
+  rpc.respond('echo', () => {
+    return new Promise((resolve) => {
+      const timeout = [...rpc._requests.values()][0].timeout
+      t.is(timeout._destroyed, false, 'sanity check')
```

Comment:
whats _destroyed? i dont like private tests like that, will be flakes in the future